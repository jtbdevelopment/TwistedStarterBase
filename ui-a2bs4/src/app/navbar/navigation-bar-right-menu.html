<ul class="navbar-nav">
    <li>
        <!--  TODO -->
        <a routerLink="/profile" *ngIf="playerLoaded" class="nav-link">
            <span class="fa fa-user"></span>
        </a>
    </li>
    <li>
        <!--  TODO -->
        <a routerLink="/help" *ngIf="playerLoaded" class="nav-link">
            <span class="fa fa-question"></span>
        </a>
    </li>
    <li>
        <!--  TODO -->
        <a routerLink="/admin" *ngIf="showAdmin" class="nav-link">
            <span class="fa fa-eye"></span>
        </a>
    </li>
    <!--  TODO -->
    <li><a href="javascript:void(0)" (click)="logout()" *ngIf="showLogout" class="nav-link">Logout</a></li>
</ul>
