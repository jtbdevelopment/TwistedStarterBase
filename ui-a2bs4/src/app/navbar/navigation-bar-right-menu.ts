import {Component, Input} from '@angular/core';

@Component({
    selector: 'navigation-bar-right-menu',
    template: require('./navigation-bar-right-menu.html')
})
export class NavigationBarRightMenuComponent {
    @Input() playerLoaded: boolean;
    @Input() showAdmin: boolean;
    @Input() showLogout: boolean;
}
