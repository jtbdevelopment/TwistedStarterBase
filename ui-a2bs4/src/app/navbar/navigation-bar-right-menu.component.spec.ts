import {inject, TestBed} from '@angular/core/testing';
import {NavigationBarRightMenuComponent} from './navigation-bar-right-menu.component';
import {RouterTestingModule} from '@angular/router/testing';
import {NgbModule, NgbPopoverConfig} from '@ng-bootstrap/ng-bootstrap';
import {HelpDisplayService} from '../help/help-display.service';
import {PlayerService} from 'jtb-core-games-ui';

class MockPlayerService {
    logout = jasmine.createSpy('logout');
}

describe('Component:  nav bar right menu component', () => {

    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [
                RouterTestingModule,
                NgbModule
            ],
            providers: [
                {provide: PlayerService, useClass: MockPlayerService},
                HelpDisplayService,
                NgbPopoverConfig
            ],
            declarations: [
                NavigationBarRightMenuComponent
            ],
        });
        TestBed.compileComponents();
    });

    it('display disabled when player not loaded', () => {
        const fixture = TestBed.createComponent(NavigationBarRightMenuComponent);
        fixture.componentInstance.playerLoaded = false;
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('displays toggle when player loaded', () => {
        const fixture = TestBed.createComponent(NavigationBarRightMenuComponent);
        fixture.componentInstance.playerLoaded = true;
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('toggling help on/off shows/hides popover', inject([HelpDisplayService], (helpService) => {
        const fixture = TestBed.createComponent(NavigationBarRightMenuComponent);
        fixture.componentInstance.playerLoaded = true;
        fixture.detectChanges();
        fixture.componentInstance.toggleHelp();
        expect(helpService.isShown()).toBeTruthy();
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
        fixture.componentInstance.toggleHelp();
        expect(helpService.isShown()).toBeFalsy();
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    }));

    it('test logout propogates to player service', inject([PlayerService], (playerService) => {
        const fixture = TestBed.createComponent(NavigationBarRightMenuComponent);
        fixture.componentInstance.logout();
        expect(playerService.logout).toHaveBeenCalledTimes(1);
    }));
});
