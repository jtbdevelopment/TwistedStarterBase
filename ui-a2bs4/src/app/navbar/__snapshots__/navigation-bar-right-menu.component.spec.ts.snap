// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:  nav bar right menu component display disabled when player not loaded 1`] = `
<app-navigation-bar-right-menu
  helpDisplay={[Function HelpDisplayService]}
  helpPopovers={[Function QueryList]}
  playerLoaded="false"
  playerService={[Function MockPlayerService]}
>
  
  
</app-navigation-bar-right-menu>
`;

exports[`Component:  nav bar right menu component displays toggle when player loaded 1`] = `
<app-navigation-bar-right-menu
  helpDisplay={[Function HelpDisplayService]}
  helpPopovers={[Function QueryList]}
  playerLoaded={[Function Boolean]}
  playerService={[Function MockPlayerService]}
>
  <ul
    class="navbar-nav"
  >
    
      
    <li>
      
          
      <a
        class="nav-link"
        href="/profile"
        ng-reflect-ngb-popover="Show your profile and stats"
        ng-reflect-placement="top"
        ng-reflect-popover-title="Profile"
        ng-reflect-router-link="/profile"
        ng-reflect-triggers="manual"
        ngbpopover="Show your profile and stats"
        placement="top"
        popovertitle="Profile"
        triggers="manual"
      >
        
              
        <span
          class="fa fa-user"
        />
        
          
      </a>
      
      
    </li>
    
      
    <li>
      
          
      <a
        class="nav-link"
        href="javascript:void(0)"
      >
        
              
        <span
          class="fa fa-question-circle"
        />
        
          
      </a>
      
      
    </li>
    
      
    <li>
      
          
      
      
      
    </li>
    
      
    <li>
      
    </li>
    
  
  </ul>
  
</app-navigation-bar-right-menu>
`;

exports[`Component:  nav bar right menu component toggling help on/off shows/hides popover 1`] = `
<app-navigation-bar-right-menu
  helpDisplay={[Function HelpDisplayService]}
  helpPopovers={[Function QueryList]}
  playerLoaded={[Function Boolean]}
  playerService={[Function MockPlayerService]}
>
  <ul
    class="navbar-nav"
  >
    
      
    <li>
      
          
      <a
        aria-describedby="ngb-popover-1"
        class="nav-link"
        href="/profile"
        ng-reflect-ngb-popover="Show your profile and stats"
        ng-reflect-placement="top"
        ng-reflect-popover-title="Profile"
        ng-reflect-router-link="/profile"
        ng-reflect-triggers="manual"
        ngbpopover="Show your profile and stats"
        placement="top"
        popovertitle="Profile"
        triggers="manual"
      >
        
              
        <span
          class="fa fa-user"
        />
        
          
      </a>
      <ngb-popover-window
        _nghost-c0=""
        class="popover bs-popover-top"
        id="ngb-popover-1"
        role="tooltip"
        style="left: 0px;"
      >
        
      
        <div
          _ngcontent-c0=""
          class="arrow"
        />
        
      
        <h3
          _ngcontent-c0=""
          class="popover-header"
        >
          Profile
        </h3>
        <div
          _ngcontent-c0=""
          class="popover-body"
        >
          Show your profile and stats
        </div>
      </ngb-popover-window>
      
      
    </li>
    
      
    <li>
      
          
      <a
        class="nav-link"
        href="javascript:void(0)"
      >
        
              
        <span
          class="fa fa-question-circle"
        />
        
          
      </a>
      
      
    </li>
    
      
    <li>
      
          
      
      
      
    </li>
    
      
    <li>
      
    </li>
    
  
  </ul>
  
</app-navigation-bar-right-menu>
`;

exports[`Component:  nav bar right menu component toggling help on/off shows/hides popover 2`] = `
<app-navigation-bar-right-menu
  helpDisplay={[Function HelpDisplayService]}
  helpPopovers={[Function QueryList]}
  playerLoaded={[Function Boolean]}
  playerService={[Function MockPlayerService]}
>
  <ul
    class="navbar-nav"
  >
    
      
    <li>
      
          
      <a
        class="nav-link"
        href="/profile"
        ng-reflect-ngb-popover="Show your profile and stats"
        ng-reflect-placement="top"
        ng-reflect-popover-title="Profile"
        ng-reflect-router-link="/profile"
        ng-reflect-triggers="manual"
        ngbpopover="Show your profile and stats"
        placement="top"
        popovertitle="Profile"
        triggers="manual"
      >
        
              
        <span
          class="fa fa-user"
        />
        
          
      </a>
      
      
    </li>
    
      
    <li>
      
          
      <a
        class="nav-link"
        href="javascript:void(0)"
      >
        
              
        <span
          class="fa fa-question-circle"
        />
        
          
      </a>
      
      
    </li>
    
      
    <li>
      
          
      
      
      
    </li>
    
      
    <li>
      
    </li>
    
  
  </ul>
  
</app-navigation-bar-right-menu>
`;
